### ISSUE TITLE
input-group-lg and input-group-sm override validation icon padding

### ISSUE THREAD

**Issue by tmorehouse** [twbs/bootstrap#30131]
Version 4.4.1 (and possibly before) Haven't tested in v5 yet.

When using `.is-valid` or `.is-invalid` on a `.form-control` in an `.input-group-lg` or `.input-group-sm`, the right padding added to accommodate for the validation icon is reset to the input group form-control default padding (non-validated), causing long text to overlap the validation icon.

As you can see in the following image, Medium (the default input-group size) has the correct righthand padding, while the small and large input-groups do not have the correct padding to accommodate for the validation icon.

![image](https://user-images.githubusercontent.com/2781561/73689353-6d446080-46a4-11ea-8c62-fb4623e63469.png)

![image](https://user-images.githubusercontent.com/2781561/73689414-8baa5c00-46a4-11ea-9839-ca19c5dead68.png)

**Comment by tmorehouse** [comment]
Perhaps the solution would be just as simple as adding an `!important` in the following at line 80:

https://github.com/twbs/bootstrap/blob/fbf294a605e36a4cf9c66f0c4637f13ded18f727/scss/mixins/_forms.scss#L75-L92

```scss
 .form-control { 
   @include form-validation-state-selector($state) { 
     border-color: $color; 
  
     @if $enable-validation-icons { 
       padding-right: $input-height-inner !important;
       // ...
     }
```

**Comment by hirayama-onagigawa-computing** [comment]
+1
I've just met the same situation.
https://jsfiddle.net/hirayama_onagigawa/xt70amuh/30/

**Comment by pricop** [comment]
+1

I'm in the same boat. Small & Large form don't get the padding treatment the regular inputs get.

**Comment by ffoodd** [comment]
[I reproduced this on v4](https://codepen.io/ffoodd/pen/GRpbzZz). After testing on v5, it's not affected only because [validation styles come after input groups](https://github.com/twbs/bootstrap/blob/master/scss/_forms.scss)—[which is not the case in v4](https://github.com/twbs/bootstrap/blob/v4-dev/scss/bootstrap.scss).

@twbs/css-review Not quite sure about the way to go with this:
1. separate form validation styles in a later imported partial?
2. moving input-group up in the imports?
3. bumping specificity for this padding, either by using `!important` or adding another piece in the selector?

Any thought?

**Comment by stefanmo** [comment]
Any fix planned for this?

**Comment by ffoodd** [comment]
Ping @twbs/css-review again :)
`!important` is easy enough, I don't expect any regression. We may avoid it by moving things as I mentionned, but would probably be a breaking change.

**Comment by ffoodd** [comment]
Just updated [my reduced test case](https://codepen.io/ffoodd/pen/GRpbzZz) to cover more things, and opened a PR to discuss this.

**Comment by XhmikosR** [comment]
Should be fixed by #32968.
